### DISCUSSION TITLE
Selecting elements

### DISCUSSION THREAD

**Discussion by abhishal76** [cypress-io/cypress#18249] [Questions and Help]
Hello, i want to know if there is any workaround for selecting elements for dynamic CSS other than adding data-* attribute?

**Comment by marktnoonan** [comment]
Hi, depends on the use case. For selecting an interactive element, I use the accessible name for that thing, say `cy.contains('button', 'Add to cart')` - avoids depending on DOM structure or classes etc. 

If it’s not interactive & there’s no specific content that’s always there to check, I really do prefer data attributes that are just for testing. Complex DOM paths or other hacks tend to cause failures often just due to refactoring a component, even if it all still works.

**Reply by abhishal76** [reply]
hmm the accessible name also changes with every new language added. Adding data attribute is an overhead for developer

**Reply by marktnoonan** [reply]
I don’t know of a general workaround to get a selector for cases where everything that identifies an element is dynamic, other than to intentionally identify it with something stable in the code for testing purposes. If you want to share more details on what specifically you’re up against I’d be happy to take a look at it though.
